man/rgbasm.5

```diff
@@ -324,9 +324,32 @@ with a zero constant as either operand will be constant 0, and
 .Sq ||
 with a non-zero constant as either operand will be constant 1, even if the other operand is non-constant.
 .Pp
-.Ic \&!
+.Sq \&!
 returns 1 if the operand was 0, and 0 otherwise.
 Even a non-constant operand with any non-zero bits will return 0.
+.Ss Integer functions
+Besides operators, there are also some functions which have more specialized uses.
+.EQ
+delim $$
+.EN
+.Bl -column "BITWIDTH(n)"
+.It Sy Name Ta Sy Operation
+.It Fn BITWIDTH n Ta Returns the number of bits necessary to represent
+.Ar n .
+Some useful formulas:
+.Ic BITWIDTH Ns Po Ar n Ns Pc - 1
+equals $\[lf] roman log2 ( n ) \[rf]$,
+.Ic BITWIDTH Ns Pq Ar n No - 1
+equals $\[lc] roman log2 ( n ) \[rc]$,
+and
+.Ic No 32 - Ic BITWIDTH Ns Pq Ar n
+equals $roman clz ( n )$.
+.It Fn TZCOUNT n Ta Returns $roman ctz ( n )$, the count of trailing zero bits at the end of the binary representation of
```

Comment:
This has some spaces between the function name and parens. Not sure how to avoid that, I'm not well-versed in `eqn`.

![Screenshot](https://github.com/user-attachments/assets/2f1d3d40-18c0-4ac7-94bf-9ced6d931fef)

Reply:
Interesting, I don't see spaces in my renderers. On Linux it simply works, no spaces, on macOS it prints floor as `|_log2(n)_|` and ceil as `|~log2(n)~|` (in keeping with other ASCIIfication, like using `` `ASCII quotes'`` instead of curly quotes or `-:-` instead of the division sign).

Reply:
If you prefer, I could change those to just "floor()" and "ceil()".

Reply:
I'm on Linux too. I'm pretty sure the difference is that you're using `groff` and I'm using `mandoc`. But I'll investigate that further tomorrow—it's a good time to learn `eqn`, after all!

Reply:
I'm using `man` which must use `groff` internally.

Reply:
Turns out this is an intentional formatting decision. Since that's decided by the typesetter, I don't think we can or ought to do anything about it.

Regardless; I'd suggest using `log sub 2` instead of `log2`, and removing the `roman` so the function names stand out more from the running text, and using `\ ` so that the expressions don't get cut by line-wrapping.

```suggestion
.Ic BITWIDTH Ns ( Ar n Ns )\ -\ 1
equals $\[lf] log sub 2 ( n ) \[rf]$,
.Ic BITWIDTH Ns Pq Ar n Ns \ -\ 1
equals $\[lc] log sub 2 ( n ) \[rc]$, and
.Ic No 32\ -\  Ns Ic BITWIDTH Ns Pq Ar n
equals $clz ( n )$.
.It Fn TZCOUNT n Ta Returns $ctz ( n )$, the count of trailing zero bits at the end of the binary representation of
```

![Screenshot of the above suggestion's rendering](https://github.com/user-attachments/assets/31472395-492c-40eb-81aa-a2753f4d6b22)

Additionally, I noticed whitespace before the commas, which actually is [a known bug in Mandoc](https://cvsweb.bsd.lv/mandoc/TODO?rev=1.334&content-type=text/x-cvsweb-markup):

> implicit whitespace around inline equations

Reply:
Why is a blatant antifeature such as force-converting everything to ASCII enabled, and why is anyone trying to work around that? The solution to that problem is to update the OS to the 21st century, not to try to fit mathematical operators into ASCII...

Reply:
Maybe that's configurable in the typesetter, but I think we hardly need to be concerned: our input *describes* our intentions well, so the typesetters can correctly interpret it once they are fixed.

Reply:
I agree with `log sub 2` and `\ ` spacing. I would argue against removing the `roman` specifier, but it's moot because if we write `log sub 2` then I think the `log` function will be recognized as the predefined macro for the logarithm function, and just like all other standard math functions (sin, cos, exp, etc) it shows up as roman. ;)  (If it didn't, it would be looking like *l* times *o* times *g*.)

Reply:
Edited the suggestion (but not the screenshots lol) to reinstate `roman` before `clz` and `ctz` so they are formatted consistently with `log` and other function names in math formulas.

Reply:
If we're nitpicking the formulas: how do you render a real minus sign?

Reply:
According to `mandoc_char(7)`, that's `\-`.